math-scala/src/main/scala/org/apache/mahout/math/scalabindings/SolverFactory.scala

```diff
@@ -0,0 +1,71 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package org.apache.mahout.math.scalabindings
+
+import java.io.File
+
+import org.apache.mahout.logging._
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime._
+import scala.reflect._
+
+
+class SolverFactory {
+
+  private final implicit val log = getLog(this.getClass)
+
+  // just temp for quick POC
+  val classMap: Map[String,String] =
+    Map(("GPUMMul"->"org.apache.mahout.viennacl.opencl.GPUMMul"),
```

Comment:
no this class may not be present, there's no direct dependency on a concrete backend. This code is correct in its intent, but we will need a Backend interface (one for VCL, one for e.g., MKL etc.) that can probe & report availability of concrete solvers.